openshift-kube-apiserver/filters/deprecatedapirequest/update_func.go

```diff
@@ -0,0 +1,111 @@
+package deprecatedapirequest
+
+import (
+	"sort"
+	"strings"
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	apiv1 "github.com/openshift/api/apiserver/v1"
+	"k8s.io/kubernetes/openshift-kube-apiserver/filters/deprecatedapirequest/v1helpers"
+)
+
+// IncrementRequestCounts add additional api request counts to the log.
+func SetRequestCountsForNode(node string, requestCounts *resourceRequestCounts) v1helpers.UpdateStatusFunc {
+	return func(status *apiv1.DeprecatedAPIRequestStatus) {
```

Comment:
musing what if.  This is not (yet) a requirement.

Can I get a `StatusRequestsLast24hToResourceRequestCountForNode(status.RequestsLast24h, nodeName) resourceRequestCounts` function and a `ResourceCountTo24HRequestLog(resourceRequestCount, NodeName) []RequestLog` that always returns 24 indexes and the top users.

then this becomes (I think)

```go
existingResourceCounts := StatusRequestsLast24hToResourceRequestCountForNode(status.RequestsLast24h, nodeName) resourceRequestCounts
requestCounts.Add(existingResourceCounts)
requiredStatusRequests = ResourceCountTo24HRequestLog(requestCounts)

```

Reply:
> musing what if. This is not (yet) a requirement.

Yeah, comparing with what is here, I think this is significantly easier.